doc: fix return results description of napi_get_value_string_utf8

taveras opened the pull request:
The API for napi_get_value_string_utf8() appears to have been
previously changed. This improves the doc reflect the current design.

Fixes: https://github.com/nodejs/node/issues/14398

##### Checklist


- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)
doc, n-api

kfarnung commented on doc/api/n-api.md:
```diff
@@ -1737,11 +1737,10 @@ napi_status napi_get_value_string_utf8(napi_env env,
-terminator. If the buffer size is insufficient, the string will be truncated
-including a null terminator.
+- `[out] result`: Number of bytes copied into the buffer, excluding the null
+terminator.
```
> If the buffer size is insufficient, the string will be truncated including a null terminator. 

I feel like this information should be represented somewhere, maybe as part of the bufsize parameter?

taveras replied:
ah, yes! do you feel it may be best to keep this sentence within the `[out] result` description?

kfarnung replied:
I think it would be better to move that sentence to be part of the `[in] bufsize` parameter comment.  That's where it's relevant to call out that the string will be truncated, but still include a null terminator.

taveras commented on doc/api/n-api.md:
```diff
@@ -1736,12 +1736,11 @@ napi_status napi_get_value_string_utf8(napi_env env,
-- `[out] result`: Number of bytes copied into the buffer including the null.
-terminator. If the buffer size is insufficient, the string will be truncated
-including a null terminator.
+- `[in] bufsize`: Size of the destination buffer. When this value is insufficient, the returned string will be truncated.
```
@kfarnung I added the details about the string truncation, but now under the `bufsize` line

jasnell replied:
nit: needs a line wrap at 80 chars :-)

taveras replied:
doh, pushed a fix 😄

mhdawson commented:
CI run: https://ci.nodejs.org/job/node-test-pull-request/9733/

mhdawson commented:
@taveras  seems like there are now conflicts, can you rebase ?

taveras commented:
@mhdawson i rebased the branch against master.

also, i added the same changes for the remaining `napi_get_value_string_*` methods,
as the original issue applied to the docs of all three methods

mhdawson commented:
@taveras thanks :)

CI run: https://ci.nodejs.org/job/node-test-pull-request/9792/

mhdawson commented:
CI failure was unrelated, opened https://github.com/nodejs/node/issues/14982

mhdawson commented:
Landing, sorry it took so long.

mhdawson commented:
Landed as 82bad0b4d8b17e998b0ff15066f253f7e47c2feb.  @taveras Thanks !
